sw/host/rom_ext_image_tools/signer/src/main.rs

```diff
@@ -21,10 +30,74 @@ fn main() {
     let config = ParsedConfig::new(&config_path);
 
     // Read raw binary.
-    let image_path = Path::new(&config.input_files.image_path);
-    let mut raw_image = RawImage::new(&image_path);
+    let mut image = RawImage::new(&config.input_files.image_path);
+
+    // Get the private key used for signature generation. It is also used to
+    // extract key public exponent and modulus.
+    let private_key_der =
+        fs::read(&config.input_files.private_key_der_path).expect("Failed to read the image!");
+
+    // Update "signed" manifest fields.
+    image.update_static_fields(&config);
+
+    let exponent = &signature_key_public_exponent_le();
+    image.update_exponent_field(exponent);
+
+    let modulus = &signature_key_modulus_le();
+    image.update_modulus_field(modulus);
+
+    // Convert ASN.1 DER private key into Mundane RsaPrivKey.
+    let private_key =
+        RsaPrivKey::parse_from_der(&private_key_der).expect("Failed to parse private key!");
+
+    // Produce the signature from concatenated system_state_value,
+    // device_usage_value and the portion of the "signed" portion of the image.
+    let image_sign_data = image.data_to_sign();
+    let device_usage_value = &device_usage_value(&config.input_files.usage_constraints_path);
+    let system_state_value = &system_state_value(&config.input_files.system_state_value_path);
+
+    let mut message_to_sign = Vec::<u8>::new();
+    message_to_sign.extend_from_slice(system_state_value);
+    message_to_sign.extend_from_slice(device_usage_value);
+    message_to_sign.extend_from_slice(image_sign_data);
+
+    let signature =
+        RsaSignature::<B3072, RsaPkcs1v15, Sha256>::sign(&private_key, &message_to_sign)
+            .expect("Failed to sign!");
+
+    image.update_signature_field(&signature.bytes());
+
+    // The whole image has been updated and signed, write the result to disk.
+    image.write_file();
+}
+
+/// Generate a dummy signature key public exponent.
+///
+/// Eventually this value will be obtained from the private key.
+fn signature_key_public_exponent_le() -> Vec<u8> {
+    vec![0xA5; 1]
+}
+
+/// Generate a dummy signature key modulus.
+///
+/// Eventually this value will be obtained from the private key.
+fn signature_key_modulus_le() -> Vec<u8> {
+    vec![0xA5; 384]
+}
```

Comment:
I feel like it might make more sense to have these be returned together as part of some RsaPrivateKey struct.

Reply:
This is a temporary dummy, I have already done a change to mundane for propper getters.